src/main/java/org/sagebionetworks/bridge/services/ParticipantFileService.java

```diff
@@ -0,0 +1,166 @@
+package org.sagebionetworks.bridge.services;
+
+import com.amazonaws.HttpMethod;
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.Headers;
+import com.amazonaws.services.s3.model.GeneratePresignedUrlRequest;
+import com.amazonaws.services.s3.model.ObjectMetadata;
+import com.amazonaws.services.s3.model.ResponseHeaderOverrides;
+import org.joda.time.DateTime;
+import org.sagebionetworks.bridge.config.BridgeConfig;
+import org.sagebionetworks.bridge.dao.ParticipantFileDao;
+import org.sagebionetworks.bridge.exceptions.BadRequestException;
+import org.sagebionetworks.bridge.exceptions.EntityAlreadyExistsException;
+import org.sagebionetworks.bridge.exceptions.EntityNotFoundException;
+import org.sagebionetworks.bridge.models.ForwardCursorPagedResourceList;
+import org.sagebionetworks.bridge.models.files.ParticipantFile;
+import org.sagebionetworks.bridge.validators.Validate;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import javax.annotation.Resource;
+
+import java.net.URL;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+
+import static com.amazonaws.HttpMethod.GET;
+import static com.amazonaws.HttpMethod.PUT;
+import static com.google.common.base.Preconditions.checkArgument;
+import static org.apache.commons.lang3.StringUtils.isNotBlank;
+import static org.sagebionetworks.bridge.BridgeConstants.API_MAXIMUM_PAGE_SIZE;
+import static org.sagebionetworks.bridge.BridgeConstants.API_MINIMUM_PAGE_SIZE;
+import static org.sagebionetworks.bridge.BridgeConstants.PAGE_SIZE_ERROR;
+import static org.sagebionetworks.bridge.validators.ParticipantFileValidator.INSTANCE;
+
+@Component
+public class ParticipantFileService {
+
+    static final int EXPIRATION_IN_MINUTES = 1440;
+
+    static final String PARTICIPANT_FILE_BUCKET = "participant-file.bucket";
```

Comment:
You also need an entry in src/main/resources/BridgeServer2.conf. (The value can be a dummy value, or a reference to a manually created org-sagebridge-participantfile-local bucket.) This creates the entry in the config, so the config system knows to look for the env var to overwrite it with.